File: aula-01/js/Ex04.js
/**
 * Gera o resto da divisão inteira de x por y
 * @param {Number} x dividendo
 * @param {Number} y divisor
 * @throws {RangeError} caso x ou y forem menores que 0
 * @returns {Number} resto da divisão de x por y
 */
module.exports = mod;
function mod(x, y) {
    if (y < 0) {
        throw new RangeError("Intervalo de y inválido");
    }

    if (x <= 0) {
        throw new RangeError("Intervalo de x inválido");
    }

    let s = x;
    while (s >= y) {
        s = s - y;
    }

    return s;
}
    

File: aula-01/java/Ex04.java
public class Ex04 {

    public static void main(String[] args) {
        System.out.println(RestoDivisao(3, 8));
    }

    public static int RestoDivisao(int x, int y) {
        
        int s;
        
        if (y < 0) {
            throw new IllegalArgumentException("Valor de y inválido");
        }

        if (x <= 0) {
            throw new IllegalArgumentException("Valor de x inválido");
        }
        
        s = x;

        while (s >= y) {
            s = s - y;
        }

        return s;
        
    }
        
}
